odl/solvers/iterative/iterative.py

```diff
@@ -290,6 +289,106 @@ def conjugate_gradient_normal(op, x, rhs, niter=1, callback=None):
             callback(x)
 
 
+def conjugate_gradient_nonlinear(f, x, rhs, niter=1, nreset=0,
+                                 line_search=None, tol=1e-16, beta_method='FR',
+                                 callback=None):
```

Comment:
I'll see what I can do. Overall our solvers could use a look so that the parameters fit better.